examples/demo_threading/rust/src/lib.rs

```diff
@@ -112,16 +112,16 @@ mod ffi {
             let sensors_changed = Arc::new(AtomicBool::new(false));
 
             // Make relevent clones so that we can pass them to the threads
-            let accumulator_sensors = Arc::clone(self.as_mut().sensors_map_mut());
+            let accumulator_sensors = Arc::clone(&unsafe { self.as_mut().rust_mut() }.sensors_map);
```

Comment:
note how this all becomes "unsafe" when it shouldn't be. And if `rust_mut()` was safe the API would still be quite nice, as it'd be saying just, I need to mutate the rust side then work on the field as normal.